services/auth/auth-supabase.service.ts

```diff
@@ -10,23 +11,34 @@ class AuthService {
     this.supabase = createClient(supabaseUrl, supabaseAnonKey);
   }
 
-  async registerUser(email: string, password: string): Promise<{ user: User | null; token: string | undefined }> {
+  private handleError(error: any): { user: null; token: undefined } {
+    errorHandler(error.message);
+    return { user: null, token: undefined };
```

Comment:
The handle error shall not return anything. but the function it self. Different functions may require different returns

Actually we may not need this function but can call`errorHandler` directly